code/game/machinery/teleporter.dm

```diff
@@ -141,10 +142,48 @@
 	for(var/obj/effect/landmark/minimap_poi/poi in SSholomap.pois)
 		if(poi.z in SSodyssey.scenario_zlevels)
 			possible_pois += poi
+	if(isemptylist(possible_pois))
+		tgui_alert(usr, "Can't find any teleport destinations.", "Teleport Error")
+
+	// actor check for id access
+	if(SSodyssey.scenario && SSodyssey.scenario.actor_accesses && !isemptylist(SSodyssey.scenario.actor_accesses))
+		// check if they have an id at all
+		var/obj/item/card/id/user_id = user.GetIdCard()
+		if(!user_id || !istype(user_id))
+			var/choice = tgui_alert(usr,
+				"Current odyssey scenario has defined ID access, but you do not seem to have an ID on you. You can get one from the actor vendor.",
+				"Teleport Warning", list("Ok", "Cancel")
+			)
+			if(choice == "Cancel")
+				return
+
+		// check if they have actor accesses set on the scenario definition
+		var/list/scenario_access_ids = list()
+		for(var/access in SSodyssey.scenario.actor_accesses)
+			var/datum/access/access_datum = access
+			scenario_access_ids += access_datum::id
+		if(isemptylist(user_id.access & scenario_access_ids))
+			var/choice = tgui_alert(usr,
+				"Current odyssey scenario has defined ID access, but you do not seem to have an ID with any such access. You can change access of your ID using the actor access terminal.",
+				"Teleport Warning", list("Ok", "Cancel")
+			)
+			if(choice == "Cancel")
+				return
+
+	// actor check for radio headset
+	if(SSodyssey.scenario && SSodyssey.scenario.radio_frequency_name)
+		var/mob/living/living = user
+		if(istype(living) && !living.check_contents_for(/obj/item/device/radio/headset/ship/odyssey))
+			var/choice = tgui_alert(usr,
+				"Current odyssey scenario has defined radio channel, but you do not seem to have a headset that can transmit that channel. You can get one from the actor vendor.",
+				"Teleport Warning", list("Ok", "Cancel")
+			)
+			if(choice == "Cancel")
+				return
```

Comment:
This isn't sufficient, someone can close the tgui and this wouldn't match, despite the intention reasonably be to cancel

Reply:
changed to if(choice != "Ok")